x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -46,7 +53,70 @@ export class DashboardToDashboardDrilldown
     return true;
   };
 
-  public readonly execute = () => {
-    alert('Go to another dashboard!');
+  public readonly execute = async (
+    config: Config,
+    context: ActionContext<VisualizeEmbeddableContract>
+  ) => {
+    const getUrlGenerator = await this.params.getGetUrlGenerator();
+    const navigateToApp = await this.params.getNavigateToApp();
+
+    const {
+      createFiltersFromRangeSelectAction,
+      createFiltersFromValueClickAction,
+    } = await this.params.getDataPluginActions();
+    const {
+      timeRange: currentTimeRange,
+      query,
+      filters: currentFilters,
+    } = context.embeddable!.getInput();
+
+    // if useCurrentDashboardFilters enabled, then preserve all the filters (pinned and unpinned)
+    // otherwise preserve only pinned
+    const existingFilters =
+      (config.useCurrentFilters
+        ? currentFilters
+        : currentFilters?.filter(f => esFilters.isFilterPinned(f))) ?? [];
+
+    // if useCurrentDashboardDataRange is enabled, then preserve current time range
+    // if undefined is passed, then destination dashboard will figure out time range itself
+    // for brush event this time range would be overwritten
+    let timeRange = config.useCurrentDateRange ? currentTimeRange : undefined;
+    let filtersFromEvent = await (async () => {
+      // TODO: not sure what would be the best way to handle types here
+      // context.data is `unknown` and comes from `EmbeddableVisTriggerContext`
+      try {
+        return (context.data as any).range
+          ? await createFiltersFromRangeSelectAction(context.data as any)
+          : await createFiltersFromValueClickAction(context.data as any);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn("DashboardToDashboard drilldown: can't extract filters from event", e);
+        return [];
+      }
+    })();
+
+    if (context.timeFieldName) {
+      const { timeRangeFilter, restOfFilters } = esFilters.extractTimeFilter(
+        context.timeFieldName,
+        filtersFromEvent
+      );
+      filtersFromEvent = restOfFilters;
+      if (timeRangeFilter) {
+        timeRange = esFilters.convertRangeFilterToTimeRangeString(timeRangeFilter);
+      }
+    }
+
+    const dashboardPath = await getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({
+      dashboardId: config.dashboardId,
+      query: config.useCurrentFilters ? query : undefined,
+      timeRange,
+      filters: [...existingFilters, ...filtersFromEvent],
+    });
+
+    const dashboardHash = dashboardPath.split('#')[1];
```

Comment:
@stacey-gammon maybe we can change the interface of URL generator while only drilldowns use it? I see two problems:

1. URL generator generates just a path part of the URL.
2. In this example we actually don't need a path, but just the hash part (see `dashboardPath.split('#')[1];`).

Maybe URL generator could return [`URL` object](https://developer.mozilla.org/en-US/docs/Web/API/URL) instead.

That way above code could be:

```ts
const url = await getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({});
const dashboardHash = url.hash;
```

Or even better:

```ts
const url = await getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({});
await url.navigate();
```

The implementer of URL generator can still just provide only path string part of the URL and URL service would construct the `URL` object.

Reply:
Right now the URL service is just static code without any dependencies on `core` and other plugins, which raises the question "Why is it a stateful plugin code instead of being a static import?".

If we add `.navigate()` method to our `URL` class it would have a runtime dependency on core's application services.

Reply:
@streamich, 

> maybe we can change the interface of URL generator while only drilldowns use it?

ML also uses it. Also some post processing is required there btw..

> Maybe URL generator could return URL object instead.

I think string is more common and something I'd expect to have from urlGenerator.  It will also be moved to the server soon, so should be common, so `navigate()` won't make sense.  Url object would also make it client specific? 

Also, please note, I pushed update with `getHref` support and left a comment. After `dashboard` app moves to np urls getHref implementation would be plain: 
getHref = (...) =>  await getUrlGenerator(...).create(...) 

_But for navigateToApp api still some "parsing" will be needed to extract just a path after appId_  ...

Reply:
> I think string is more common and something I'd expect to have from urlGenerator.

You would still get the path string, it would just be

```ts
await getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({}).path
```

instead of

```ts
await getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({})
```

> Url object would also make it client specific?

No, `URL` class can be isomorphic.

Reply:
Still think having it as a string is most common and the simplest (imho). 
If consumer needs parsed version then it is a one line: `url.parse(href)`
> navigate 

navigate is still pure client thing, right?. There are also a bunch of options depending on how you want to navigate? using's core's service? or maybe open in a new tab? Also it means that every `generator` will have to implement that.

Reply:
> navigate is still pure client thing, right?

I guess it will be only client-side.

> Also it means that every `generator` will have to implement that.

Why? I think generators would not be changed, they would still implement the same interface they implement now. We don't need to change generators. For now, we would just change return value of `.createUrl()` from `string` to `{ path: string }`.

Reply:
@streamich , if to extend url generator, I'd leave `createUrl` as is to return a `string`, but maybe added 
createUrlObject (or createLocation(): Location) which would return `url.parse(_url)`